src/App.tsx

```diff
@@ -1,24 +1,48 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
+type State = {
+  isClockVisible: boolean,
+  clockName: number,
+};
 
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
+class App extends React.Component<{}, State> {
+  state: State = {
+    isClockVisible: true,
+    clockName: Math.random(),
+  };
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+  showTime = () => {
+    this.setState({
+      isClockVisible: true,
+    });
+  };
+
+  hideTime = () => {
+    this.setState({
+      isClockVisible: false,
+    });
+  };
+
+  changeName = () => {
+    this.setState({
+      clockName: Math.random(),
+    });
+  };
+
+  render(): React.ReactNode {
+    return (
+      <div className="container">
+        {this.state.isClockVisible && (
+          <Clock name={this.state.clockName} />
+        ) }
+        <button type="button" className="btn btn-outline-success" onClick={this.showTime}>Show Clock</button>
```

Comment:
If there are 3 or more attributes, write each attribute on the new line

```suggestion
        <button
       	  type="button"
       	  className="btn btn-outline-success"
       	  onClick={this.showTime}
       	>
       	  Show Clock
       	</button>
```